citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -47,229 +47,266 @@
-                results.setCitations(localPageRequest.getCitations());
-            }
-            
-            return results;
-        }
-        throw new GroupDoesNotExistException("There is no group with id " + groupId);
+    public CitationResults getGroupItems(IUser user, String groupId, String collectionId, int page, String sortBy,
+	    String conceptTag) throws GroupDoesNotExistException {
+	Optional<CitationGroup> groupOptional = groupRepository.findById(new Long(groupId));
+	if (groupOptional.isPresent()) {
+	    ICitationGroup group = groupOptional.get();
+
+	    List<PageRequest> requests = null;
+	    try {
+		if (conceptTag == null) {
+		    requests = pageRequestRepository
+			    .findByUserAndObjectIdAndPageNumberAndZoteroObjectTypeAndSortByAndCollectionId(user,
+				    groupId, page, ZoteroObjectType.GROUP, sortBy, collectionId);
+		} else {
+
+		    if (collectionId == null) {
+			requests = pageRequestRepository.findPageRequestWithObjectIdAndConceptTags(user, groupId, page,
+				ZoteroObjectType.GROUP, conceptTag, sortBy);
+
+		    } else {
+			requests = pageRequestRepository.findPageRequestWithObjectIdAndCollectionIdAndConceptTags(user,
+				groupId, page, ZoteroObjectType.GROUP, conceptTag, sortBy, collectionId);
+
+		    }
+		}
+
+	    } catch (JpaObjectRetrievalFailureException ex) {
+		logger.warn("Could not retrieve page request.", ex);
+	    }
+
+	    IPageRequest localPageRequest = null;
+	    if (requests != null && requests.size() > 0) {
+		// there should be just one
+		localPageRequest = requests.get(0);
+
+		OffsetDateTime updatedOn = localPageRequest.getLastUpdated() != null ? localPageRequest.getLastUpdated()
+			: OffsetDateTime.MIN;
+		OffsetDateTime groupLastLocallyModified = group.getLastLocallyModifiedOn() != null
+			? group.getLastLocallyModifiedOn()
+			: OffsetDateTime.MIN;
+		if (updatedOn.plusMinutes(syncFrequencey).isAfter(OffsetDateTime.now())
+			&& updatedOn.isAfter(groupLastLocallyModified)) {
+		    CitationResults results = new CitationResults();
+		    results.setCitations(
+			    localPageRequest.getCitations().stream().distinct().collect(Collectors.toList()));
+		    results.getCitations().sort(new Comparator<ICitation>() {
+			@Override
+			public int compare(ICitation o1, ICitation o2) {
+			    return sortFunctions.get(sortBy).apply(o1, o2);
+			}
+		    });
+		    results.setTotalResults(group.getNumItems());
+		    return results;
+		}
+	    }
+
+	    CitationResults results = new CitationResults();
+	    results.setCitations(new ArrayList<>());
+	    results.setTotalResults(0);
+	    results.setNotModified(true);
+//	    if (collectionId == null || collectionId.trim().isEmpty()) {
+//		results = zoteroManager.getGroupItems(user, groupId, page, sortBy, group.getVersion());
+//	    } else {
+//		results = zoteroManager.getCollectionItems(user, groupId, collectionId, page, sortBy,
+//			group.getVersion());
+//	    }
```

Comment:
if not needed, remove

Reply:
actually, you removed the code that would get an update from zotero. why? that doesn't make sense. also the next line will never be true since you set notModified to true.